src/core/engines/cross-margin/CrossMarginMath.sol

```diff
@@ -173,31 +182,50 @@ library CrossMarginMath {
         }
     }
 
+    /**
+     * @notice get numeraire and underlying needed to fully collateralize
+     * @dev calculates left side and right side of the payout profile
+     * @param putStrikes array of put option strikes
+     * @param putWeights amount of options at a given strike
+     * @param hasCalls has call options
+     * @param pois are the strikes we are evaluating
+     * @param payouts are the payouts at a given strike
+     * @return numeraireNeeded with {collateral asset's} decimals
+     * @return underlyingNeeded with {underlying asset's} decimals
+     */
```

Comment:
same comment here: Not sure what would happen if collateral asset == underlying asset

Reply:
see above